builtin/readline_osh.py

```diff
@@ -42,19 +46,57 @@ def __exit__(self, type, value, traceback):
         if self.orig_keymap_name is not None:
             self.readline.restore_orig_keymap()
 
+class BindXCallback(object):
+    """A callable we pass to readline for executing shell commands."""
+
+    def __init__(self, eval):
+        # type: (Eval) -> None
+        self.eval = eval
+
+    def __call__(self, cmd, line_buffer, point):
+        # type: (str, str, int) -> Tuple[int, str, str]
+        """Execute a shell command through the evaluator.
+
+        Args:
+          cmd: The shell command to execute
+          line_buffer: The current line buffer
+          point: The current cursor position
+        """
+        print("Setting READLINE_LINE to: %s" % line_buffer)
+        print("Setting READLINE_POINT to: %s" % point)
+        print("Executing cmd: %s" % cmd)
+
+        # TODO: add READLINE_* env vars later
+        # assert isinstance(line_buffer, str)
+        # self.mem.SetNamed(location.LName("READLINE_LINE"),
+        #                  value.Str(line_buffer),
+        #                  scope_e.GlobalOnly,
+        #                  flags=SetExport)
+        
+        # TODO: refactor out shared code from Eval, cache parse tree?
+        
+        cmd_val = cmd_eval.MakeBuiltinArgv([cmd])
+        status = self.eval.Run(cmd_val)
+
+        return (status, line_buffer, str(point))
+
 
 class Bind(vm._Builtin):
     """Interactive interface to readline bindings"""
 
-    def __init__(self, readline, errfmt):
-        # type: (Optional[Readline], ui.ErrorFormatter) -> None
+    def __init__(self, readline, errfmt, mem, evaluator):
```

Comment:
It doesn't seem like `mem` is used now

Reply:
Yes, but I think I'll need it for the READLINE_* env vars.